react/private/withTextProps.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import omit from 'lodash/omit';
+import forEach from 'lodash/forEach';
+
+export const sizes = [
+  'small',
+  'substandard',
+  'superstandard',
+  'standard',
+  'subheading',
+  'headline',
+  'heading',
+  'hero'
+];
+
+const getBooleanSizes = () => {
+  const booleanProps = {};
+
+  forEach(sizes, size => {
+    booleanProps[size] = PropTypes.bool;
+  });
+};
+
+export const SizePropTypes = {
+  size: PropTypes.oneOf(sizes),
+  ...getBooleanSizes()
+};
+
+const parseBooleanSize = props => {
+  const sizeProps = {};
+
+  forEach(sizes, size => {
+    if (props[size]) {
+      sizeProps.size = size;
+    }
+  });
+
+  return sizeProps;
+};
+
+const withTextProps = OriginalComponent => {
+  const DecoratedComponent = props => {
+    const newProps = {
+      ...parseBooleanSize(props),
+      ...omit(props, sizes)
```

Comment:
The order of these should be flipped I think, if `props` contained `size` as well as one of `sizes`, it would override the result of `parseBooleanSize` which is not desired right?

Reply:
Depends on what we want to achieve here and which ones are more important. Could just add a proptype check that either boolean or size should be supplied and not both

Reply:
Because technically that's not the usecase

Reply:
Added custom proptypes checker :)